### ISSUE TITLE
InstallBasicPackageFiles: FIRST_TARGET may be useless

### ISSUE THREAD

**Issue by drdanz** [robotology/ycm#215]
Using this mini-project (with CMake 3.13.4, I still have to ckeck older CMake versions)

```cmake
cmake_minimum_required(VERSION 3.5)
project(foo)

add_library(foo foo.cpp)

install(TARGETS foo
        EXPORT FOO
        ARCHIVE DESTINATION lib
        LIBRARY DESTINATION lib)
export(EXPORT FOO
       FILE FooTargets.cmake)
```

The target file contains

```cmake
# Protect against multiple inclusion, which would fail when already imported targets are added once more.
set(_targetsDefined)
set(_targetsNotDefined)
set(_expectedTargets)
foreach(_expectedTarget foo)
  list(APPEND _expectedTargets ${_expectedTarget})
  if(NOT TARGET ${_expectedTarget})
    list(APPEND _targetsNotDefined ${_expectedTarget})
  endif()
  if(TARGET ${_expectedTarget})
    list(APPEND _targetsDefined ${_expectedTarget})
  endif()
endforeach()
if("${_targetsDefined}" STREQUAL "${_expectedTargets}")
  unset(_targetsDefined)
  unset(_targetsNotDefined)
  unset(_expectedTargets)
  set(CMAKE_IMPORT_FILE_VERSION)
  cmake_policy(POP)
  return()
endif()
if(NOT "${_targetsDefined}" STREQUAL "")
  message(FATAL_ERROR "Some (but not all) targets in this export set were already defined.\nTargets Defined: ${_targetsDefined}\nTargets not yet defined: ${_targetsNotDefined}\n")
endif()
unset(_targetsDefined)
unset(_targetsNotDefined)
unset(_expectedTargets)
```

This protects from multiple inclusion of the target file. Therefore it might not be necessary to add the check in the config file before including the target file, we could just include it, and CMake will do the rest.
This makes it completely useless to have the `FIRST_TARGET` argument.

**Comment by claudiofantacci** [comment]
This is very interesting. I find the `FIRST_TARGET` quite difficult to digest and to document in the `InstallBasicPackageFiles`. It would be a plus to get rid of it 👍

**Comment by drdanz** [comment]
I totally agree!
Before removing it I want to make some tests, though.

**Comment by traversaro** [comment]
The "Protect against multiple inclusion..." comment was present since CMake 2.8.12, so I think we are good  to go:
https://github.com/Kitware/CMake/commit/4b1919f73e7663799285171b472d76053fb99166

**Comment by PeterBowman** [comment]
I recall struggling with this option because of https://github.com/robotology/ycm/issues/148, not sure if removing it would negatively affect current code somewhere (I'll check that later).

**Comment by drdanz** [comment]
@PeterBowman I'm not planning to *remove* it, I'm planning to mark it as deprecated, and passing it will just print a deprecated warning, and do nothing.

**Comment by drdanz** [comment]
BTW, the issue of not being able to read targets from the `EXPORT`, mentioned in https://github.com/robotology/ycm/issues/148#issuecomment-397408775 is probably no longer an issue.

**Comment by drdanz** [comment]
Fixed in #236

**Comment by claudiofantacci** [comment]
Awesome 🎉 

Just to understand this correctly. We can **just** remove the `TARGET`/`TARGETS`/`FIRST_TARGET` provided that we used `EXPORT` for the target. Is it right?
